synthesizer/process/src/lib.rs

```diff
@@ -131,20 +167,50 @@ impl<N: Network> Process<N> {
     /// Adds a new stack to the process.
     /// If you intend to `execute` the program, use `deploy` and `finalize_deployment` instead.
     #[inline]
-    pub fn add_stack(&mut self, stack: Stack<N>) {
+    pub fn add_stack(&self, stack: Stack<N>) {
         // Add the stack to the process.
-        self.stacks.insert(*stack.program_id(), Arc::new(stack));
+        let mut lock = self.stacks.lock();
+        match lock {
+            Ok(ref mut lock) => lock.put(*stack.program_id(), Arc::new(stack)),
+            Err(_) => None,
+        };
     }
 }
 
 impl<N: Network> Process<N> {
+    /// Initializes a new process.
+    /// Assumption: this is only called in test code.
+    #[inline]
+    pub fn load_testing_only() -> Result<Self> {
+        Process::load_from_storage(Some(aleo_std::StorageMode::Development(0)))
+    }
+
     /// Initializes a new process.
     #[inline]
-    pub fn load() -> Result<Self> {
-        let timer = timer!("Process::load");
+    pub fn load_from_storage(storage_mode: Option<StorageMode>) -> Result<Self> {
+        let timer = timer!("Process::load_testing_only");
+
+        debug!("Opening storage");
+        let storage_mode = storage_mode.as_ref().ok_or_else(|| anyhow!("Failed to get storage mode"))?.clone();
```

Comment:
```suggestion
        let storage_mode = storage_mode.cloned().ok_or_else(|| anyhow!("Failed to get storage mode"))?;
```
if the compiler doesn't complain for some reason, this is simpler

Reply:
893a17194